### PR TITLE
CI: upgrade ARM64 runner and github-cli orb

### PR DESCRIPTION
**Pull request by kleisauke**
See:
https://discuss.circleci.com/t/linux-image-deprecations-and-eol-for-2024/50177
https://circleci.com/docs/using-arm/#ubuntu-2004---focal

### CODE REVIEW

**Comment by lovell** [comment]
Thanks, I made this change on the sharp repo earlier too (CircleCI sent a rather verbose email about it).

**Comment by kleisauke** [comment]
I also received an email about this. Commit eeacc819b4ef3c6c42a02106c65deef898f8489b also updates the [`circleci/github-cli`](https://circleci.com/developer/orbs/orb/circleci/github-cli) orb while we are here.

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,4 +23,4 @@ jobs:
       - when:
           condition: <<pipeline.git.tag>>
           steps:
-            - run: gh release upload --repo lovell/sharp-libvips $CIRCLE_TAG *.tar.gz *.integrity
+            - run: gh release upload --repo "$(git config --get remote.origin.url)" $CIRCLE_TAG *.tar.gz *.integrity
```
Comment (kleisauke) [inline .circleci/config.yml:26]: Note to self: I considered using the new [`/upload`](https://circleci.com/developer/orbs/orb/circleci/github-cli#jobs-upload) job here, e.g.:
```suggestion
            - github-cli/upload:
                files: "*.tar.gz *.integrity"
```

But that job seems to do a `gh repo clone`, so it probably won't work on the recently build files.
https://github.com/CircleCI-Public/github-cli-orb/blob/e89b4c51027a6659a54235d0d3358f7ff1145494/src/jobs/upload.yml#L45

**Comment by lovell** [comment]
Dankjewel Kleis. I spotted some deprecation warnings from the checkout action too, silenced via https://github.com/lovell/sharp-libvips/commit/4720684c736dc204987cebd1e5abbecaddae2632

**Pull request merged**